src/tokens/Achievements/Achievements.cairo

```diff
@@ -0,0 +1,286 @@
+        range_check_ptr
+    }() -> (owner: felt):
+    let (owner: felt) = Ownable.owner()
+    return (owner)
+end
+
+@view
+func paused{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }() -> (paused: felt):
+    let (paused) = Pausable.is_paused()
+    return (paused)
+end
+
+#
+# Externals
+#
+
+@external
+func setURI{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(uri: felt):
+    Ownable.assert_only_owner()
+    ERC1155._set_uri(uri)
+    return ()
+end
+
+@external
+func setApprovalForAll{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(operator: felt, approved: felt):
+    Pausable.assert_not_paused()
+    ERC1155.set_approval_for_all(operator, approved)
+    return ()
+end
+
+@external
+func safeTransferFrom{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        from_: felt,
+        to: felt,
+        id: Uint256,
+        amount: Uint256,
+        data_len: felt,
+        data: felt*
+    ):
+    Pausable.assert_not_paused()
+    ERC1155.safe_transfer_from(from_, to, id, amount, data_len, data)
+    return ()
+end
+
+
+@external
+func safeBatchTransferFrom{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        from_: felt,
+        to: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*,
+        data_len: felt,
+        data: felt*
+    ):
+    Pausable.assert_not_paused()
+    ERC1155.safe_batch_transfer_from(
+        from_, to, ids_len, ids, amounts_len, amounts, data_len, data)
+    return ()
+end
+
+@external
+func mint{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        to: felt,
+        id: Uint256,
+        amount: Uint256,
+        data_len: felt,
+        data: felt*
+    ):
+    Ownable.assert_only_owner()
+    let (caller) = get_caller_address()
+    ERC1155._mint(to, id, amount, data_len, data)
+    return ()
+end
+
+@external
+func mintBatch{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        to: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*,
+        data_len: felt,
+        data: felt*
+    ):
+    Ownable.assert_only_owner()
+    let (caller) = get_caller_address()
+    ERC1155._mint_batch(to, ids_len, ids, amounts_len, amounts, data_len, data)
+    return ()
+end
+
+@external
+func burn{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(from_: felt, id: Uint256, amount: Uint256):
+    Ownable.assert_only_owner()
+    let (caller) = get_caller_address()
+    with_attr error_message("ERC1155: called from zero address"):
+        assert_not_zero(caller)
+    end
```

Comment:
```suggestion
```

I think we can remove this since we are asserting only owner which asserts this internally